### PR TITLE
installer: exit with 1 if euid != 0

### PR DESCRIPTION
**Pull request by GiedriusS**
Running void-installer only makes sense if you run it as root so exit with 1 if euid != 0. I got caught by this a few times when I ran `void-installer` on live iso, on void user and only after going through a few menus I realize that I haven't ran it as root :(

### CODE REVIEW

File: installer.sh.in [hunk 1]
```diff
@@ -1006,6 +1006,12 @@ if [ ! -x /bin/dialog ]; then
     echo "ERROR: missing dialog command, exiting..."
     exit 1
 fi
+
+if [ "$(id -u)" != "0" ]; then
```
Comment (xtraeme) [inline installer.sh.in]: The correct way to compare a number would be `-ne 0`, this way there's no need to quote.

Comment (GiedriusS) [inline installer.sh.in]: I agree. Added another commit that drops these quotes.

**Comment by chneukirchen** [comment]
Sorry, but that's bs... the "" are required in case id fails and returns an empty string.

**Comment by GiedriusS** [comment]
You are right. Removed the last commit.

**Comment by Duncaen** [comment]
`"$(id -u)" -ne 0`

**Comment by GiedriusS** [comment]
I think that would fail if `id -u` returned an empty string. Try in dash, for example:
`[ "" -ne 0 ] && echo "it works"`

**Pull request closed without merging**